hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/client/DefaultCertificateClient.java

```diff
@@ -111,8 +112,10 @@ public abstract class DefaultCertificateClient implements CertificateClient {
 
   private static final Random RANDOM = new SecureRandom();
 
-  private static final String CERT_FILE_NAME_FORMAT = "%s.crt";
+  public static final String CERT_FILE_NAME_FORMAT = "%s.crt";
+  public static final String CA_CERT_PREFIX = "CA-";
   private static final int CA_CERT_PREFIX_LEN = 3;
+  public static final String ROOT_CA_CERT_PREFIX = "ROOTCA-";
```

Comment:
On the master branch there is a CAType enum now that contains these values. I suggest using that enum, and redefining `CA_CERT_PREFIX_LEN` as `CAType.SUBORDINATE.getFileNamePrefix().length()` and `ROOT_CA_PREFIX_LEN` as `CAType.ROOT.getFileNamePrefix().length()` .

Reply:
@Galsza , thanks for the review. testUgi  is necessary to make sure OM can connect SCM regarding kerberos login and kerberos service authentication.  testSecureFlag for DefaultCAServer is to support a more fine grained certificate lifetime. CAType is a good suggestion. I will refactor this part of code.